soc/arc/snps_nsim/smp.c

```diff
@@ -12,11 +12,9 @@
  *
  */
 
-#include <zephyr/device.h>
 #include <zephyr/init.h>
-#include "soc.h"
+#include <zephyr/arch/arc/v2/arc_connect.h>
 
-#ifdef CONFIG_SMP
 static int arc_nsim_init(const struct device *dev)
```

Comment:
if this indeed moves to a module called smp.c, then let's not call this arc_nsim_init (which made sense for soc.c), but something smp specific.

Reply:
exactly! That is why my preference is not to change anything and just leave it as soc.c, this sys_init stuff really belongs into soc.c and moving things around like this for the purpose of cleaning up (the PR title still says "remove soc.h usage") and this change has nothing to do with soc.h usage and now we are getting into renaming stuff etc etc....

Oh well...

Reply:
> exactly! That is why my preference is not to change anything and just leave it as soc.c, this sys_init stuff really belongs into soc.c and moving things around like this for the purpose of cleaning up (the PR title still says "remove soc.h usage") and this change has nothing to do with soc.h usage and now we are getting into renaming stuff etc etc....
> 
> Oh well...

I'm not sure how such reviewing behavior helps to make progress. It's nice to talk about consistency, but it's also nice to help to make progress on PRs that precisely help towards that goal. Can we please focus on the technical aspects and comment on latest changes? I see no comments related to [`31e1a9f` (#46585)](https://github.com/zephyrproject-rtos/zephyr/pull/46585/commits/31e1a9f2a47c7afb037b70fd6df08f0751193c2b)

Reply:
My comment to not only rename the filename but also the function name (arc_nsim_init) to make it SMP specific was about this commit. There are many improvements to get rid of soc.h in this PR, but avoiding the sometimes empty soc.c meets different opinions. If you prefer to include this commit, please rename arc_nsim_init, e.g. to arc_nsim_smp_init.

Reply:
> My comment to not only rename the filename but also the function name (arc_nsim_init) to make it SMP specific was about this commit. There are many improvements to get rid of soc.h in this PR, but avoiding the sometimes empty soc.c meets different opinions. If you prefer to include this commit, please rename arc_nsim_init, e.g. to arc_nsim_smp_init.

done

Reply:
Thanks, I'm ok with this.